### DISCUSSION TITLE
Dealing with Liveview stream

### DISCUSSION THREAD

**Discussion by heru** [woylie/flop_phoenix#174] [Ideas]
hi @woylie,

is there any suggestion how to handle flop phoenix with liveview streams ?

**Comment by woylie** [comment]
The table component requires a small update to be able to accept a stream instead of a list. For the state handling, I don't have any insights yet. Once I've had the time to explore this, I'll add an example to the docs.

**Comment by woylie** [comment]
I opened a draft PR with the necessary changes. I added some comments you may find helpful. https://github.com/woylie/flop_phoenix/pull/176

**Comment by heru** [comment]
Hi,

Thanks for quick response.

I think it is worthed in a specific case. Especially when you have to using
liveview with thousand rows of data and infinite scrolling table is not the
option for such situation.

Thanks.

**Comment by type1fool** [comment]
I have a few Flop tables with streams, and everything is working well except for empty streams. I tried adding the `:no_results_content` opt, as well as removing opts from the table completely, but the empty row doesn't render when the `@items` is a stream.

In a non-flop table, I have this code to handle empty lists and streams:

```html
<tr
  :if={@rows == [] or (is_struct(@rows, LiveStream) and @rows.inserts == [])}
  id="table-empty-row"
>
  <% colspan = if @actions, do: length(@col) + 1, else: length(@col) %>
  <td colspan={colspan} class="bg-gray-50 text-center align-middle py-4 rounded-b-lg">
    No records
  </td>
</tr>
```

I would be happy to open a PR for this if you'd like.

**Reply by woylie** [reply]
Hi @type1fool, a PR would be appreciated!

**Reply by type1fool** [reply]
Alright, I'll set a reminder to work on this.
